public/content/en/vibed/deploy-on-heroku.md

```diff
@@ -0,0 +1,186 @@
+- Your application should compile without error using dmd (`dub build`). 
+
+### 1 : Setup the app
+
+Heroku needs to know how to communicate with a deployed application. 
+Hence a global `PORT` environment variable is provided which needs to be injected into an application and 
+it should bind and listen to this port. 
+For development a default port (here __8080__) should be set:
+
+```d
+shared static this() {
+  // ...
+  auto settings = new HTTPServerSettings;
+  // Provide a default port in case of the $PORT variable isn't set.  
+  settings.port = environment.get("PORT", "8080").to!ushort;
+  listenHTTP(settings, router);
+}
+```
+
+Additionally create a `Procfile`, which is a text file in the root directory of the application, which explicitly declare what command 
+should be executed to start the app.
+
+The `Procfile` in the example app looks like this:
+
+```
+web: ./hello-world
+```
+
+### 2 : Prepare the app 
+
+Before going further login to the Heroku Command Line by using the [Heroku Toolbelt](https://toolbelt.heroku.com/standalone).
+
+This provides access to the Heroku Command-Line Interface (CLI), which can be used for managing and scaling your applications and add-ons.
+
+After installing the toolbet run the following
+
+```
+$ heroku login
+```
+
+### 3 : Create the app 
+
+Go on the [heroku dashboard](https://dashboard.heroku.com) and create a new app. 
+After doing this memorize the name of the created app, it will be useful later. 
+
+or use the Command-Line like this 
+
+```
+$ heroku create
+Creating app... done, ⬢ rocky-hamlet-67506
+https://rocky-hamlet-67506.herokuapp.com/ | https://git.heroku.com/rocky-hamlet-67506.git
+```
+
+The app's name here is *rocky-hamlet-67506*. 
+
+### Deploy using git 
+
+To deploy the app directly use `git` commands. A separate git remote endpoint should be added to which new releases can be pushed.
+Thus the name of the newly created application 
+in the previous chapter needs to be passed as argument .(here it is rocky-hamlet-67506):
+
+```
+$ heroku git:remote -a rocky-hamlet-67506
+```
+
+Notice the remote endpoint is added to the git config
+
+```
+$ git remote -v
+heroku	https://git.heroku.com/rocky-hamlet-67506.git (fetch)
+heroku	https://git.heroku.com/rocky-hamlet-67506.git (push)
+```
+
+### Adding the buildpack
+
+Buildpacks are responsible for generate assets or compiled code.
+
+For more information browse the [Heroku documentation](https://devcenter.heroku.com/articles/buildpacks)
+
+For deployement the [Vibe.d buildpack](https://github.com/MartinNowak/heroku-buildpack-d) can be used : 
+
+```
+$ heroku buildpacks:set https://github.com/MartinNowak/heroku-buildpack-d
+```
+By default the buildpack uses the latest dmd compiler. 
+It is possible to use GDC or LDC and to choose a specific compiler versions by adding a `.d-compiler` file to your project. 
+
+Use `dmd`, `ldc`, or `gdc` to select the latest or dmd-2.068.2, ldc-0.16.0, or gdc-4.9.2 to 
```

Comment:
I would put dmd-2.0xxx etc. into backticks too because it is a value used in a configuration